tests/capture/test_capture_cond.py

```diff
@@ -516,3 +611,88 @@ def test_circuit_consts(self, pred, arg, expected):
         jaxpr = jax.make_jaxpr(circuit_with_consts)(*args)
         res_ev_jxpr = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, *args)
         assert np.allclose(res_ev_jxpr, expected), f"Expected {expected}, but got {res_ev_jxpr}"
+
+    @pytest.mark.parametrize("reset", [True, False])
+    @pytest.mark.parametrize("postselect", [None, 0, 1])
+    @pytest.mark.parametrize("shots", [None, 20])
+    def test_mcm_predicate_execution(self, reset, postselect, shots):
+        """Test that QNodes executed with mid-circuit measurement predicates for
+        qml.cond give correct results."""
+        device = qml.device("default.qubit", wires=3, shots=shots, seed=jax.random.PRNGKey(1234))
+
+        def true_fn(arg):
+            qml.RX(arg, 0)
+
+        def false_fn(arg):
+            qml.RY(arg, 0)
+
+        @qml.qnode(device)
+        def f(x, y):
+            qml.RX(x, 0)
+            m = qml.measure(0, reset=reset, postselect=postselect)
+
+            qml.cond(m, true_fn, false_fn)(y)
+            return qml.expval(qml.Z(0))
+
+        params = [2.5, 4.0]
+        res = f(*params)
+        qml.capture.disable()
+        expected = f(*params)
+
+        assert np.allclose(res, expected), f"Expected {expected}, but got {res}"
+
+    @pytest.mark.parametrize("shots", [None, 100])
+    @pytest.mark.parametrize(
+        "params, expected",
+        [
+            ([np.pi / 2, 0, 0, np.pi / 2], (0, 0, 0, 0)),  # true_fn
+            ([0, np.pi / 2, 0, np.pi / 2], (0, 0, -1, 0)),  # elif_fn1
+            ([0, 0, np.pi / 2, np.pi / 2], (1 / (2 * np.sqrt(2)), 0, -0.5, 0.5)),  # elif_fn2
+            ([0, 0, 0, 0], (1 / np.sqrt(2), 0, 0, 1)),  # false_fn
```

Comment:
I found it hard to trace those cases in my head, and what is happening to the state/MCMs throughout, until the terminal measurement. The rotation angles lead to equal superpositions, mixing the branches, right?
Is there an easy way you could comment on what is happening?

Reply:
I'll add some more comments.

Reply:
I might be overlooking something here, but doesn't `RX(np.pi/2,0)` create a superposition state of comp. basis states? i.e., it's not a `PauliX`, is it?
```python
>>> qml.RX(np.pi/2, 0).matrix()
array([[0.70710678+0.j        , 0.        -0.70710678j],
       [0.        -0.70710678j, 0.70710678+0.j        ]])
```